odoo/addons/base/models/res_users.py

```diff
@@ -191,10 +191,11 @@ def _search_full_name(self, operator, operand):
                 sub_where = expression.AND([group_domain, category_domain])
             else:
                 sub_where = expression.OR([group_domain, category_domain])
-            if operator in expression.NEGATIVE_TERM_OPERATORS:
-                where = expression.AND([where, sub_where])
-            else:
-                where = expression.OR([where, sub_where])
+            where.append(sub_where)
+        if operator in expression.NEGATIVE_TERM_OPERATORS:
+            where = expression.AND(where)
+        else:
+            where = expression.OR(where)
```

Comment:
Please avoid using the same variable for two different things: `where` is a list of domains, then a domain.
I suggest to rename `where` above as `where_domains`, and to rewrite this as
```suggestion
        if operator in expression.NEGATIVE_TERM_OPERATORS:
            return expression.AND(where_domains)
        else:
            return expression.OR(where_domains)
```